great_expectations/data_context/types/base.py

```diff
@@ -1173,7 +1173,23 @@ class DataContextConfigSchema(Schema):
     )
     config_variables_file_path = fields.Str(allow_none=True)
     anonymous_usage_statistics = fields.Nested(AnonymizedUsageStatisticsConfigSchema)
-    concurrency = fields.Nested(ConcurrencyConfigSchema)
+    concurrency = fields.Nested(
+        ConcurrencyConfigSchema, required=False, allow_none=True
+    )
+
+    # To ensure backwards compatability, we need to ensure that new options are "opt-in"
+    # If a user has not explicitly configured the value, it will be None and will be wiped by the post_dump hook
+    REMOVE_KEYS_IF_NONE = [
+        "concurrency",  # 0.13.33
+    ]
+
+    @post_dump
+    def remove_keys_if_none(self, data: dict, **kwargs) -> dict:
+        data = copy.deepcopy(data)
+        for key in self.REMOVE_KEYS_IF_NONE:
+            if key in data and data[key] is None:
+                data.pop(key)
+        return data
 
```

Comment:
This is a placeholder for the time being - if we can get to a place where all nulls are wiped (rather that specifying specific fields), I think this would be much more maintainable.

Reply:
Agreed. Also, this is probably fine here for now, but given similar serialization issues with other configs (e.g. the expectation_context issue for Expectation Suites), if we decide that this method is the right way to go, may want to make this method available as a utility outside of this specific class.

Reply:
I'm not sure if we can move this function out from the class tbh. The `@post_dump` decorator needs to be on a class method and this particular method uses class state (int he form of `self.REMOVE_KEYS_IF_NONE`.

I have pretty much copied this from another schema so I'd love to hear your thoughts about how to make this reusable.

Reply:
Ah - interesting. Sounds like marshmallow magic with which I'm not too familiar. Let's leave as-is for now.